hearth/lib/hearth/http/middleware/content_md5.rb

```diff
@@ -16,8 +16,14 @@ def initialize(app, _ = {})
         def call(input, context)
           request = context.request
           unless request.headers.key?('Content-MD5')
+            context.logger.debug(
```

Comment:
This information is basically the same in every middleware (and is pretty minimal) - instead of needing to add log statements to every one, could we do it through the middleware stack (eg: the app gets wrapped with a logger that knows the class, and which would just log the start/end but otherwise pass through).  Or alternatively, a middleware base class?

Reply:
Edit:  in some of the middlewares there is more detailed/specific logging, but it does seem like the start/end of the middleware would be easy to have the stack log instead.

Reply:
I think it's different per middleware. I am essentially wrapping the meat of the middleware, like for Build I am wrapping the actual build part, not the start and end of middleware execution. It's also conscious of interceptor invokes. We could do one or the other.

Reply:
Yeah - thats fair.  The specialized per middleware is likely more useful.

Another thought - we only log in interceptor invoke when actual interceptors are run, not at the hook point.  Maybe worth logging that as well since its the core lifecycle of an SRA request?

Reply:
I'm not sure either .. I am leaning towards keeping it as is, because the debug logging is really useful for logic that's actually executing.